src/AgFEM/CellAggregation.jl

```diff
@@ -53,47 +55,61 @@ function aggregate(
   geo::CSG.Geometry,
   in_or_out)
 
+  trian = Triangulation(cut,geo)
+  cell_to_cut_meas = cell_measure(trian,num_cells(cut.bgmodel))
   cell_to_inoutcut = compute_bgcell_to_inoutcut(cut,geo)
   facet_to_inoutcut = compute_bgfacet_to_inoutcut(cut_facets,geo)
-  aggregate(strategy,cut.bgmodel,in_or_out,cell_to_inoutcut,facet_to_inoutcut)
+  aggregate(strategy,cut.bgmodel,in_or_out,cell_to_cut_meas,cell_to_inoutcut,facet_to_inoutcut)
 end
 
-function aggregate(strategy,model::DiscreteModel,in_or_out,cell_to_inoutcut,facet_to_inoutcut)
+function aggregate(
+  strategy,model::DiscreteModel,in_or_out,cell_to_cut_meas,cell_to_inoutcut,facet_to_inoutcut)
   @abstractmethod
 end
 
-struct AggregateAllCutCells end
+struct AggregateCutCellsByThreshold
+  threshold::Float64
+  AggregateCutCellsByThreshold(x) =
+    ( x < 0.0 ) || ( x > 1.0 ) ? error("Agg. threshold must be in [0,1]") : new(x)
+end
+
+struct AggregateAllCutCells
+  AggregateAllCutCells() = AggregateCutCellsByThreshold(1.0)
+end
 
 function aggregate(
-  strategy::AggregateAllCutCells,model::DiscreteModel,in_or_out,cell_to_inoutcut,facet_to_inoutcut)
-  _compute_aggregates(cell_to_inoutcut,facet_to_inoutcut,model,in_or_out)
+  strategy::AggregateCutCellsByThreshold,model::DiscreteModel,
+  in_or_out,cell_to_cut_meas,cell_to_inoutcut,facet_to_inoutcut)
+  _compute_aggregates(
+    strategy.threshold,cell_to_cut_meas,cell_to_inoutcut,facet_to_inoutcut,model,in_or_out)
 end
 
-function _compute_aggregates(cell_to_inoutcut,facet_to_inoutcut,model,loc)
+function _compute_aggregates(threshold,cell_to_cut_meas,cell_to_inoutcut,facet_to_inoutcut,model,loc)
   @assert loc in (IN,OUT)
   trian = get_triangulation(model)
+  cell_to_meas = cell_measure(trian,num_cells(model))
+  cell_to_unit_cut_meas = cell_to_cut_meas./cell_to_meas
```

Comment:
Minor thing:

This allocates a new global array. We can avoid it by using a lazy array
```julia
cell_to_unit_cut_meas = apply(/,cell_to_cut_meas,cell_to_meas)
```

Reply:
Done in 891a11d